plugins/ebitdo/fu-ebitdo-device.c

```diff
@@ -512,8 +512,14 @@ fu_ebitdo_device_write_firmware (FuDevice *device,
 	/* when doing a soft-reboot the device does not re-enumerate properly
 	 * so manually reboot the GUsbDevice */
 	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
-	if (!g_usb_device_reset (usb_device, error)) {
```

Comment:
In other plugins there are a few error codes that we ignore, i.e. we do the error_local thing. I think it would be wise to do this here, as even in the non-FWUPD_DEVICE_FLAG_WILL_DISAPPEAR case a reset failing is quite a common thing due to a hub in the chain doing slightly the wrong thing.

I think it also makes sense to move the `fu_device_set_remove_delay` bit out either before or after the reset too.

Reply:
> In other plugins there are a few error codes that we ignore, i.e. we do the error_local thing. I think it would be wise to do this here, as even in the non-FWUPD_DEVICE_FLAG_WILL_DISAPPEAR case a reset failing is quite a common thing due to a hub in the chain doing slightly the wrong thing.

Are you just saying that usb reset failing shouldn't be fatal no matter it disappears or not?

Can you give an example?  I looked through the tree some and didn't see anything.

> 
> I think it also makes sense to move the `fu_device_set_remove_delay` bit out either before or after the reset too.

This is explicitly only for the case of `will-disappear`.  So if we make usb reset non-fatal all the time, then it still needs that guard.